pkg/client/keycloak/config.go

```diff
@@ -57,9 +57,21 @@ func (c *KeycloakRealmConfig) setDefaultURIs(baseURL string) {
 	c.TokenEndpointURI = baseURL + "/auth/realms/" + c.Realm + "/protocol/openid-connect/token"
 }
 
+func (kc *KeycloakConfig) GetSSOProviderRealm() *KeycloakRealmConfig {
+	provider := kc.SelectSSOProvider
+	switch provider {
+	case MAS_SSO:
+		return kc.KafkaRealm
+	case REDHAT_SSO:
+		return kc.RedhatSSORealm
+	default:
+		return kc.KafkaRealm
+	}
+}
```

Comment:
Why this method? We already have this: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/c99eb533406db7e1e7e6b2c437eb617087895df5/pkg/services/sso/keycloak.go#L27

Reply:
In some places, we have keycloakconfig direct references to get URLs current. For now, use this method & later refactor them to use GetRealmConfig() *keycloak.KeycloakRealmConfig  via keycloak service.

Reply:
I believe we are accumulating too many of these 'for now'. Can we check if in those places where we have keycloakconfig we can't use the keycloakservice directly?

Reply:
I'll create a jira to identify all the places & try to replace them with a keycloak service.

Reply:
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/connector/internal/handlers/authentication.go#L29
Here don't make sense to inject service

Reply:
It make sense [here](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/connector/internal/handlers/authentication.go#L14)
Then [here](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/connector/internal/handlers/authentication.go#L29) you can use:
```
KeysURL(keycloakService.GetRealmConfig().JwksEndpointURI) 
```

Reply:
*authentication.HandlerBuilder: type sso.KeycloakService does not exist in the container

It will require a lot of changes to implement for injecting it with di

Reply:
I believe you have to use `sso.KafkaKeyKloakService`.
Dunno why, but it has been remapper: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/7f22d08f523b76f072147a06ab028ed07f68c014/pkg/providers/core.go#L84

Reply:
Even if you provide `sso.KafkaKeyKloakService.` di expect the authentication handler to implement the keycloak service. Replacing with Keycloak config with the keycloak service is also going to provide the same config object. I don't see any issues. One benefit would be we get the config of the selected provider. but there could be a chance we need to set multiple jwks URLs configured in keycloak config irrespective of the provider configured.

Reply:
It's just a matter of duplicate code. We already have logic to switch between KeycloakService and RedhatSSOService and here we are duplicating the same logic.
If someone will need to fix a bug, add a provider, etc, it will be natural to update the builder, while with high probability no-one will check here unless they intentionally search for it. 
Anyway, mine was just a suggestion. I'm ok if you don't agree.

Reply:
your suggestion is valid, I was thinking about some other use-cases.